iree/tools/CMakeLists.txt

```diff
@@ -380,6 +388,12 @@ iree_cc_test(
     iree::vm::variant_list
 )
 
+if(CMAKE_CROSSCOMPILING)
+  iree_build_host_executable(iree-opt)
```

Comment:
we've been keeping everything nice and declarative and this breaks that by not respecting the global settings (IREE_BUILD_COMPILER, etc) and instead makes an imperative command ("build this host executable now" vs. "declare this host executable exists"). I should be able to cross-compile without requiring iree-run-mlir or iree-opt at all, for example.

Reply:
Oh, it might be the name being misleading. The function under the hood just declares a bunch of custom targets/commands for `iree-opt`. So if no targets are really depending on `iree-opt`, it won't be built. So it's not that we are "building this host executable now" at here; it actually means "declare this host executable exists". Let me rename the function to avoid the confusion.

Reply:
renamed.